packages/flutter/lib/src/cupertino/app.dart

```diff
@@ -334,38 +226,15 @@ class _CupertinoAppState extends State<CupertinoApp> {
   }
```

Comment:
you need the logic from the MaterialApp for changing the hero controller on changing navigator key here too

Reply:
Done